fake-lmp-device-register

```diff
@@ -23,46 +28,79 @@ class Options(NamedTuple):
     production: bool
 
 
-def create_key(uuid: str, factory: str, production: bool) -> Tuple[bytes, bytes]:
-    key = subprocess.check_output(
-        ["openssl", "ecparam", "-genkey", "-name", "prime256v1"]
-    )
-
-    with NamedTemporaryFile() as cnf:
-        cnf.write(
-            f"""[req]
-prompt = no
-distinguished_name = dn
-req_extensions = ext
-
-[dn]
-CN={uuid}
-OU={factory}
-""".encode()
-        )
-        if production:
-            cnf.write(b"businessCategory=production\n")
-        cnf.write(
-            b"""
-[ext]
-keyUsage=critical, digitalSignature
-extendedKeyUsage=critical, clientAuth
-"""
-        )
-        cnf.flush()
+def create_key(uuid: str, factory: str, production: bool, sota_dir: str) -> Tuple[bytes, bytes]:
+    try:
+        log.info("Setting path for the private key file: %s", sota_dir)
+        keyfile_path = os.path.join(sota_dir, "pkey.pem")
 
-        r = subprocess.run(
-            ["openssl", "req", "-new", "-config", cnf.name, "-key", "-"],
-            input=key,
-            stdout=subprocess.PIPE,
+        log.info("Creating key for UUID: %s, Factory: %s", uuid, factory)
+        subprocess.check_call(
+            ["openssl", "ecparam", "-genkey", "-name", "prime256v1", "-out", keyfile_path]
         )
-        r.check_returncode()
 
-    return key, r.stdout
+        with open(keyfile_path, "rb") as keyfile:
+            key_data = keyfile.read()
+
+        log.info("Creating CSR for UUID: %s, Factory: %s", uuid, factory)
+        with NamedTemporaryFile() as cnf:
+            cnf.write(
+                f"""[req]
+    prompt = no
+    distinguished_name = dn
+    req_extensions = ext
+
+    [dn]
+    CN={uuid}
+    OU={factory}
+    """.encode()
+            )
+            if production:
+                cnf.write(b"businessCategory=production\n")
+            cnf.write(
+                b"""
+    [ext]
+    keyUsage=critical, digitalSignature
+    extendedKeyUsage=critical, clientAuth
+    """
+            )
+            cnf.flush()
+
+            log.info("Running openssl req")
+            r = subprocess.run(
+                ["openssl", "req", "-new", "-config", cnf.name, "-key", keyfile_path],
+                input=key_data,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+            )
+            r.check_returncode()
+
+    except subprocess.CalledProcessError as e:
+        log.error("A subprocess error occurred: %s", e)
+        log.error("Command output: %s", e.output.decode())
+        raise
+
+    except FileNotFoundError as e:
+        log.error("File not found: %s", e)
+        raise
+
+    except PermissionError as e:
+        log.error("Permission denied: %s", e)
+        raise
```

Comment:
what value are any of these logs adding? The CalledProcessError might be helpful, but you could also just let subprocess send stderr to stderr and skip that.

Reply:
I see your point. 
I think what I wanted to address here is done in https://github.com/foundriesio/factory-registration-ref/pull/19/files#diff-512de4dd01e9bed57aa07e405bc17c7656b99b8c8ba2fd549ac90a7aa543e972R69-R71

So, we can close this one.